contract/src/lib.rs

```diff
@@ -66,12 +66,12 @@ impl Crossword {
         /* delete old function call key*/
         Promise::new(env::current_account_id()).delete_key(answer_pk);
     }
-    pub fn claim_reward(&mut self, receiver_acc_id: String, memo: String) {
+    pub fn claim_reward(&mut self, crossword_pk: Base58PublicKey, receiver_acc_id: String, memo: String) {
         let signer_pk = env::signer_account_pk();
         /* check to see if signer_pk is in the puzzles keys */
         let puzzle = self
             .puzzles
-            .get_mut(&signer_pk)
+            .get_mut(&PublicKey::from(crossword_pk.clone()))
```

Comment:
```suggestion
            .get_mut(&crossword_pk.0)
```

Don't need to clone just to take a reference to the pk. `PublicKey` is an alias to `Vec<u8>` and the inner type of `Base58PublicKey` is also `Vec<u8>`.

There should definitely be a better interface for this, but this should work for the current version